ee/app/livechat-enterprise/server/hooks/checkDefaultAgentOnNewRoom.js

```diff
@@ -0,0 +1,36 @@
+import { callbacks } from '../../../../../app/callbacks';
+import { RoutingManager } from '../../../../../app/livechat/server/lib/RoutingManager';
+import { settings } from '../../../../../app/settings';
+import { LivechatRooms, LivechatVisitors, Users } from '../../../../../app/models/server';
+
+callbacks.add('livechat.checkDefaultAgentOnNewRoom', (defaultAgent, defaultGuest) => {
```

Comment:
since this callback is ran from community code, it should be added only if an appropriate license is applied

Reply:
@sampaiodiego yeah, I totally agree. This is the reason I moved all of the [enterprise callback imports](https://github.com/RocketChat/Rocket.Chat/pull/17581/files#diff-5e738e16fa408203ea23bf5b6f809541R1) to the [onLicense callback event in this PR](https://github.com/RocketChat/Rocket.Chat/pull/17581/files#diff-4d0461894c842a1ee975d9261f153bcdR23).

Unfortunately that PR has not been merged yet, even it seems to be ready/approved.

Reply:
> @sampaiodiego yeah, I totally agree. This is the reason I moved all of the [enterprise callback imports](https://github.com/RocketChat/Rocket.Chat/pull/17581/files#diff-5e738e16fa408203ea23bf5b6f809541R1) to the [onLicense callback event in this PR](https://github.com/RocketChat/Rocket.Chat/pull/17581/files#diff-4d0461894c842a1ee975d9261f153bcdR23).
> 
> Unfortunately that PR has not been merged yet, even it seems to be ready/approved.

oh, nice.. (my comment was second comment was added during review, I didn't see your reply before adding this second comment).

looks like PR #17581 is not approved though

Reply:
@sampaiodiego I just submitted a new commit changing the way the callback event handlers are added/removed, based on the setting state.

https://github.com/RocketChat/Rocket.Chat/pull/17666/files#diff-3c816b0b79b99ce7388d5e62e21d408fR1-R88